### DISCUSSION TITLE
RPi zero (not W or 2W) and no 80 column card

### DISCUSSION THREAD

**Discussion by cineguerrilha** [tjboldt/Apple2-IO-RPi#182] [General]
hi, I know I was stupid but I rushed in this build after finding the pcb in pcbway and rushing to build it without reading the instructions properly... I was hoping there was some magic trick that would make it to work... I have an apple 2 withouth the 80 column card and i am using the rasp 0 w (no wifi) connected to the internet using an usb cable. The computer starts with the RPi splash, waiting for pi, dot dot dot, OK and then nothing. I went up to item 11, and I also tried the adding 2 drives part with apple2.io.hdv image and another I wv got from apple2,org. Any ideas wwill be very welcome

**Comment by tjboldt** [comment]
Since you see the "Waiting for RPi" means you have the firmware correctly on the card... The "OK" after that means it communicated with the RPi and the apple2driver service is running. Can you ssh into the RPi and see if there are any errors in the syslog?

`cat /var/log/syslog`

A successful boot of the Apple II should look like this:

```
Aug 20 09:47:56 apple2iorpi apple2driver[324]: Reset request
Aug 20 09:47:56 apple2iorpi apple2driver[324]: Read block 0000 in slot 5, drive 1...succeeded
Aug 20 09:47:56 apple2iorpi apple2driver[324]: Read block 0002 in slot 5, drive 1...succeeded
```

**Reply by cineguerrilha** [reply]
Thanks a lot for the reply! I cannot ssh to it as the RPi USB network stop working after installing the apple2 drive. But removing the card from the Pi I copied a part of the syslog:

Aug 20 04:11:54 apple2 systemd[1]: systemd-update-utmp-runlevel.service: Succeeded.
Aug 20 04:11:54 apple2 systemd[1]: Finished Update UTMP about System Runlevel Changes.
Aug 20 04:11:54 apple2 systemd[1]: Startup finished in 4.363s (kernel) + 55.433s (userspace) = 59.796s.
Aug 20 04:11:54 apple2 apple2driver[236]: Read block 0000 in slot 7, drive 1...succeeded
Aug 20 04:11:54 apple2 apple2driver[236]: Reset request
Aug 20 04:11:54 apple2 apple2driver[236]: Read block 0000 in slot 7, drive 1...succeeded
Aug 20 04:11:54 apple2 apple2driver[236]: Read block 0000 in slot 7, drive 1...succeeded
Aug 20 04:11:54 apple2 apple2driver[236]: Reset request
Aug 20 04:11:54 apple2 apple2driver[236]: Read block 0000 in slot 7, drive 1...succeeded
Aug 20 04:11:54 apple2 apple2driver[236]: Read block 0000 in slot 7, drive 1...succeeded
Aug 20 04:11:54 apple2 apple2driver[236]: Reset request

couldd it be the apple2-rpi.hdv file? Again, thanks a lot for making this possible!

richardson

**Comment by tjboldt** [comment]
It looks like it successfully reads block 0000 which contains the boot loader to get ProDOS going but it never goes beyond that. So either block 0000 is corrupt in the image or there is some other communication issue. The thing is, I'd expect the Apple II to crash if a corrupt block is loaded and executed. Is it just hanging at the `OK`?

**Reply by cineguerrilha** [reply]
yes, it just freezes in the ok, control reset, goes back to the rom basic. those repeated resets happen without pressing reset in the apple, for some reason it reads the block and resets.

**Reply by cineguerrilha** [reply]
hi Again, just bought the pi zero 2w and the 80 column card but the same is happening, it freezes during after the ...OK (pi found). Also, the pi zero stops connecting to wifi after the appe2-io-rpi is installed using the installation script (classic setup). Btw, i am a new to programming eeproms so I didn't quite get the firmware at $Cs00. When do I need to worry with that? Does it mean that need to offset the at28c64.bin data by $C700 if the card is in slot 7?

**Reply by tjboldt** [reply]
The firmware on the AT28C64B just fills the whole device... the addressing to $Cx00 is done by the logic chips on the card when being addressed by the Apple II. I think your firmware is probably fine since it successfully sent a command to the apple2driver software on the RPi Zero 2 W. I think the hard drive image might be incorrect. You can also have the apple2driver generate a drive image in memory which is the default.
